src/row_codec.rs

```diff
@@ -60,6 +60,27 @@ pub(crate) struct SstRowEntry {
     pub value: ValueDeletable,
 }
 
+impl PartialEq<RowEntry> for SstRowEntry {
+    fn eq(&self, other: &RowEntry) -> bool {
+        let full_key = if self.key_prefix_len > 0 {
+            let mut full_key = other.key.slice(..self.key_prefix_len).to_vec();
```

Comment:
This is a little tricky. If we had two `RowEntry`s with keys "foobar" and "bazbar", and an SstRowEntry with `prefix_len=3` and `key_suffix=bar`, both `RowEntry`s would be equal. I suppose this isn't _wrong_, per-se. It's a subtle behavior, though. 🤔 Seems worth documenting using `///` at the very least.

Reply:
Good catch, something like this ?
```text
/// Implements equality comparison between `SstRowEntry` and `RowEntry`.
/// 
/// This implementation considers two `RowEntry` instances with different keys but the same suffix
/// and an `SstRowEntry` with a prefix length and matching suffix as equal.
/// For example, if we had two `RowEntry`s with keys "foobar" and "bazbar", and an `SstRowEntry`
/// with `prefix_len=3` and `key_suffix=bar`, both `RowEntry`s would be considered equal to the `SstRowEntry`.
```

Reply:
Sounds good! I also pinged Discord to solicit feedback from others on this functionality since it's a little 😵‍💫

Reply:
i think it might better to map `SstRowEntry` to `RowEntry` before making comparisions in tests.

there're no guarantee to ensure these two structs (one is for on-disk data, and the other is in-memory) will always have same fields set, i am afraid that we may introduce some fields that not exists in both side in the future 🤔.

Reply:
That makes sense to me. Doing so would require traversing the previous keys to assemble a full key for the `SstRowEntry`, right? In that case, perhaps we should get rid of this `PartialEq` all together, and instead have the `decode_codec_entries` call `restore_full_key` so we can do an apples-to-apples comparison on RowEntries in the test.